.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI
+
+on:
+  push:
+    branches:
+      - 'trunk'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: false  # Let each trunk commit finish a build
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [8, 11, 17, 21]
+    name: Compile and Check Java ${{ matrix.java }}
+    steps:
+      - name: Setup Gradle
+        uses: ./.github/actions/setup-gradle
+        with:
+          java-version: ${{ matrix.java }}
+          gradle-cache-read-only: false
+      - name: Compile and validate
+        run: ./gradlew --build-cache --info --scan check -x test
+
+  test:
+    needs: validate
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [ 11, 17 ]
+    name: JUnit tests Java ${{ matrix.java }}
+    steps:
+      - name: Setup Gradle
+        uses: ./.github/actions/setup-gradle
+        with:
+          java-version: ${{ matrix.java }}
+          gradle-cache-read-only: false
+      - name: Test
+        timeout-minutes: 180  # 3 hours
+        run: |
+          ./gradlew --build-cache --scan --continue \
+          -PtestLoggingEvents=started,passed,skipped,failed \
+          -PignoreFailures=true -PmaxParallelForks=2 \
+          -PmaxTestRetries=1 -PmaxTestRetryFailures=10 \
```

Comment:
Maybe we don't need retries in CI? That can helps us to find out more flaky

Reply:
I agree, but this is how Jenkins is run today. Let's save this for a future PR :)